src/game/GameWorldLoader.ts

```diff
@@ -0,0 +1,300 @@
+ *
+ * @param gameWorld The data definition of the game world
+ * @param random A function used for generating random values between 0 and 1
+ * @returns A Game<Params> object which uses the given data
+ */
+export function load(
+    gameWorld: GameWorld,
+    random: () => number = Math.random,
+): Game<Params> {
+    const defaultParams = {
+        flags: gameWorld.defaultState.flags,
+        vars: gameWorld.defaultState.state,
+    }
+    const cards = gameWorld.cards.map<Card<Params>>((data) =>
+        cardFromData(data, defaultParams),
+    )
+    const eventCards = eventCardsFromData(gameWorld.eventCards, defaultParams)
+    const events: StateModifier<Params>[] = gameWorld.events.map((event) => {
+        const card = eventCards[event.initialEventCardId]
+        return eventFromData(event, card, random)
+    })
+    const parameterCaps = parameterCapsFromStats(gameWorld.stats)
+    const stats = statsFromData(gameWorld.stats)
+    const stateExtensions = stateExtensionsFromData(
+        gameWorld.worldStateModifiers,
+    )
+    return new BasicGame<Params>([...cards], stats, defaultParams, {
+        tickModifiers: [...events, ...stateExtensions, parameterCaps],
+        random,
+    })
+}
+
+/**
+ * Creates a StateModifier representing an event using data from GameWorld
+ *
+ * @param event The event data
+ * @param card The card to which the event should be linked
+ * @param random The random function used for the probability check
+ * @returns An event trigger in the form of a StateModifier
+ */
+function eventFromData(
+    event: GameWorld['events'][number],
+    card: Card<Params>,
+    random: () => number,
+): StateModifier<Params> {
+    const paramQueries = event.isAvailableWhen.map(worldQueryToParamQuery)
+    return (state) => {
+        const noPreviousCardSetter = !state.card
+        const propabilityHit = random() <= event.probability
+        const shouldExecute =
+            noPreviousCardSetter &&
+            propabilityHit &&
+            hasMatchingParamQuery(state.params, paramQueries)
+        return shouldExecute
+            ? {
+                  ...state,
+                  card: card,
+              }
+            : state
+    }
+}
+
+/**
+ * Creates a list of Stat<Params> from GameWorld data
+ *
+ * @param stats The stats data
+ * @returns A list of Stats
+ */
+function statsFromData(stats: GameWorld['stats']): Stat<Params>[] {
+    return stats.map<Stat<Params>>((stat) => ({
+        ...stat,
+        getValue: ({ params }) => params.vars[stat.id] ?? 0,
+    }))
+}
+
+/**
+ * Creates the standard parameters caps used for a standard GameWorld
+ *
+ * @param stats The stats data
+ * @returns A stat modifier which caps all the vars matched to ids in stats to [0, 100]
+ */
+function parameterCapsFromStats(stats: GameWorld['stats']) {
+    const statVarIds = stats.map((stat) => stat.id)
+    return createParameterCap(statVarIds, 0, 100)
```

Comment:
Really nice to allow customizing the range like this.

Could this be changed between scenarios by the stats definitions coming from data?

1-100 works for now as a default range, but we might want to create an issue for the future where we allow each stat to define it's own custom range. This would make it possible to implement things like a global carbon budget for example, ticking away at a rate that you can influence. Just an idea.

the `StatDefinition` `ContentType` could be extended with `min: number` and `max: number`